librato/__init__.py

```diff
@@ -420,6 +421,89 @@ def list_alerts(self, active_only=True, **query_props):
             return alerts
 
     #
+    # Spaces!
+    #
+    def list_spaces(self, **query_props):
+        """List all spaces"""
+        resp = self._mexe("spaces", query_props=query_props)
+        return self._parse(resp, "spaces", Space)
+
+    def get_space(self, id, **query_props):
+        """Get specific space by ID"""
+        resp = self._mexe("spaces/%s" % id,
+                          method="GET", query_props=query_props)
+        return Space.from_dict(self, resp)
+
+    def find_space(self, name, **query_props):
+        """Find specific space by Name"""
+        spaces = self.list_spaces(query_props=query_props)
+        # Find the Space by name
+        space = [s for s in spaces if s.name == name][0]
+        # Now use the ID to hydrate the space attributes (charts)
+        return self.get_space(space.id, query_props=query_props)
+
+    def update_space(self, space, **query_props):
+        """Update an existing space"""
+        payload = space.get_payload()
```

Comment:
This will fail as `get_payload()` returns the charts and the api does not accept
that parameter. Maybe remote it from the payload before making the requests?